### DISCUSSION TITLE
Best way to sell to a restricted group of users?

### DISCUSSION THREAD

**Discussion by stapelberg** [pretix/pretix#2726] [Q&A] [answered]
Typically, pretix events are open to the public.

But, for a series of events (one per week), I need to restrict access to only a limited group of people. Ideally, the shop will not be accessible unless with a user/password combination.

Additionally, for each event, there should be a maximum of one order per person (or possibly two per person, we’ll see).

How can I model this with pretix? I didn’t see a feature to create user accounts for folks. Instead, the email address seems to be asked in each order. Perhaps we could make pretix get that email address from the HTTP basic auth username instead, and thereby move the authentication specifics outside of pretix?

Or would I need to work with vouchers perhaps to also model the n-orders-per-person limit? 

I’m thankful for any tips :)

**Comment by raphaelm** [comment, marked as answer]
Obviously, yes, you could put some auth in front of pretix, but I also believe we have three decent options for various nuances of the problem, one of which is using vouchers:
https://docs.pretix.eu/en/latest/user/events/guides/restricted_audience.html

**Reply by stapelberg** [reply]
Brilliant! I couldn’t find this guide before asking my question, but this looks exactly like what I need :)

Thanks!

**Comment by choseh** [comment]
I don't want to raise an new issue or start an extra discussion, but from all the experience I've had with pretix for several months now, it would really be a lifesaver, if you could add an option to allow vouchers to be redeemed outside of the presale timeframe. 

We're having events where there's a subset of people that are "set" via a list that need the ability to sign up before the public can. Right now our only way is to set the ticket budget to the amount of issued vouchers, and then in the second phase, raise the ticket budget (Kontingent) to the final amount of participants. Adding an option to allow vouchers redemptions before the presale start date would make it a lot easier for all the events i'm currently supporting.

@raphaelm Do you think this might be possible in the future?

**Reply by raphaelm** [reply]
This is not the first request for it, but it can already be solved by using different products or product variations and setting the time constraints on them instead of the full event. Im not fully decided, but I'm so far not convinced it is worth the additional complexity (both in code and in user interface) it would bring.

**Reply by choseh** [reply]
Thanks. So basically you would create like a "reserved Ticket" product that's only available with a voucher, and another that starts selling later?

**Reply by choseh** [reply]
> This is not the first request for it, but it can already be solved by using different products or product variations and setting the time constraints on them instead of the full event. Im not fully decided, but I'm so far not convinced it is worth the additional complexity (both in code and in user interface) it would bring.

user interface wise, it would imho suffice to add a checkbox to the vouchers that says "allow redemption outside of presale" or something. code wise, I obviously cannot judge

**Reply by raphaelm** [reply]
> Thanks. So basically you would create like a "reserved Ticket" product that's only available with a voucher, and another that starts selling later?

yes

**Reply by choseh** [reply]
Actually works like a charm, and I think this is an acceptable solution :)
